tasks/main.yml

```diff
@@ -2,6 +2,43 @@
 - name: Setup firewalld
   include_tasks: firewalld.yml
 
+- name: Collect service facts
+  service_facts:
+  when: firewall_disable_conflicting_services | bool
+
+- name: Attempt to stop and disable conflicting services
+  service:
+    name: "{{ item }}"
+    state: stopped
+    enabled: false
+  loop: "{{ __firewall_conflicting_services }}"
+  when:
+    - firewall_disable_conflicting_services | bool
+    - (
+        ansible_facts.service_mgr == "systemd" and
+        item + ".service" | string in ansible_facts.services
+      )
+        or
+      (
+        ansible_facts.service_mgr != "systemd" and
+        item | string in ansible_facts.services
+      )
+    - (
+        ansible_facts.service_mgr == "systemd" and
+        ansible_facts.services[item + ".service"]["status"] == "enabled"
+      )
+        or
+      (
+        ansible_facts.service_mgr != "systemd" and
+        ansible_facts.services[item]["status"] == "enabled"
+      )
```

Comment:
Changed the conditions for disabling conflicting processes to specifically check the service manager, and only append ".service" if running on systemd.  Does this seem like a sufficient implementation for the check @richm?

Reply:
I think it would be cleaner by using an intermediate var, and I don't think checking for state is necessary:
```
- name: Attempt to stop and disable conflicting services
  service:
    name: "{{ item }}"
    state: stopped
    enabled: false
  loop: "{{ __firewall_conflicting_services }}"
  vars:
    __service_name: "{{ (ansible_facts.service_mgr == "systemd") |
      ternary(item ~ '.service', item) }}"
  when:
    - firewall_disable_conflicting_services | bool
```
Isn't the `service` module idempotent?  Or is it a purely performance issue to check explicitly first?

Reply:
service module fails if the service is not installed when disabling is attempted in either check or diff mode.

Reply:
Recent push implemented intermediate var for the service name.